### ISSUE TITLE
Hyperparameters 

### ISSUE THREAD

**Issue by julianikulski** [julianikulski/data-science-resources#10]
**Resource**: https://yashuseth.blog/2019/06/12/bert-explained-faqs-understand-bert-working/
**Description**: BERT Explained – A list of Frequently Asked Questions
**Details**: This blog post includes the hyperparamters used in the original BERT academic paper

**Resource**: https://arxiv.org/pdf/2101.06326.pdf
**Description**: Grid Search Hyperparameter Benchmarking of BERT, ALBERT, and LongFormer on DuoRC
**Details**: Academic paper includes comparisons of hyperparameters used for different models

**Resource**: https://pytorch-lightning.readthedocs.io/en/stable/hyperparameters.html#lightningmodule-hyperparameters
**Description**: Hyperparameters
**Details**: PyTorch Lightning docs on hyperparameters

**Resource**: https://skorch.readthedocs.io/en/stable/user/quickstart.html
**Description**: Quickstart of skorch library
**Details**: Docs of skorch library show brief example of using grid search with neural nets

**Resource**: https://towardsdatascience.com/skorch-pytorch-models-trained-with-a-scikit-learn-wrapper-62b9a154623e
**Description**: SKORCH: PyTorch Models Trained with a Scikit-Learn Wrapper
**Details**: Blog post with examples on how to use skorch with PyTorch

**Resource**: https://medium.com/optuna/using-optuna-to-optimize-pytorch-lightning-hyperparameters-d9e04a481585
**Description**: Using Optuna to Optimize PyTorch Lightning Hyperparameters
**Details**: Short tutorial in Towards Data Science on the usage of Optuna with PyTorch Lightning for hyperparameter tuning

**Resource**: https://github.com/PyTorchLightning/pytorch-lightning/issues/3233
**Description**: auto_scale_batch_size not working with datamodule #3233
**Details**: Explanation of how to use auto_scale_batch_size in PyTorch Lightning when datamodule is implemented separately and passed to trainer.fit() separately. This searches the highest batch size which fits into memory.

**Resource**: https://pytorch-lightning.readthedocs.io/en/latest/lr_finder.html
**Description**: Learning Rate Finder PyTorch Lightning 
**Details**: Description in PyTorch Lightning docs how to use the built-in tuner to determine the best initial learning rate
